### PR TITLE
OpenGL requirement on Mac OS X

### PR DESCRIPTION
**Pull request by antonioiuliano2**
In OS X include folders are different (see https://en.wikibooks.org/wiki/OpenGL_Programming/Installation/Mac), thus leading to the builder assuming you do not have the required package.

This edit should solve the issue.
Waiting for your comments.
Best regards,
Antonio

### CODE REVIEW

File: opengl.sh [hunk 1]
```diff
@@ -6,6 +6,9 @@ system_requirement_missing: |
     * On Ubuntu-compatible systems you probably need: libglu1-mesa-dev
 system_requirement: ".*"
 system_requirement_check: |
-  printf "#include <GL/glu.h>\n" | cc -xc - -c -o /dev/null
+  case $ARCHITECTURE in
```
Comment (DaniilSu) [inline opengl.sh]: I suggest to do the OS check in the following way:

if [ `uname` = Darwin ]; then
  printf "#include <OpenGL/glu.h>\n" | cc -xc - -c -o /dev/null
else
  printf "#include <GL/glu.h>\n" | cc -xc - -c -o /dev/null
fi

Using above written lines the one wouldn't need to extra specify $ARCHITECTURE variable during the installation

Comment (antonioiuliano2) [reply opengl.sh]: I agree, it seems a good idea to me.
Antonio

**Comment by ThomasRuf** [comment]
Hi Antonio,
did you implemented the suggestion of Daniil?
thomas

**Comment by antonioiuliano2** [comment]
Dear Thomas,

Last year I opened that pull request and the concerning discussion for
Martina, who needed to install FairShip on her Mac.
As far as I remember, Daniil's suggestion worked smoothly, but then the
installation got stuck for other issues.
But for what concerns this pull request, it should be fine. To be sure, I
am forwarding this mail to Martina, who might remember better the details.

Not having a Mac myself, I cannot check.

Best regards,
Antonio

Il giorno lun 29 lug 2019 alle ore 11:54 Thomas Ruf <
notifications@github.com> ha scritto:

**Comment by antonioiuliano2** [comment]
Dear Antonio, Thomas,

my apologies for the late reply. It is correct, since that advice provided us with the solution to the specific issue raised by the installation on my local machine.

Unfortunately, there were further incompatibilities which didn't allow for a complete installation of FairShip. In the meanwhile I have also changed laptop and upgraded the OS, so I guess we could try to adapt alibuild to the newest version of macOS. As already pointed out to Oliver some time ago, I am available to test/help on that side.

Best regards,
 Martina


________________________________

**Comment by ThomasRuf** [comment]
what should I do with this pull request? reject it, until updated recipe for Mac appears, or merge it?

thomas

**Comment by antonioiuliano2** [comment]
I have committed Daniil's suggestion.

For my opinion, this specific condition should be general enough to work also in updated MacOs. I have asked a colleague using Mac Mojave to launch uname and it returned Darwin.

Best regards,
Antonio Iuliano

**Pull request merged**